src/Builder/PackagesBuilder.php

```diff
@@ -136,17 +136,49 @@ private function pruneIncludeDirectories()
             $pattern = '#^' . str_replace('%hash%', '([0-9a-zA-Z]{' . strlen($hash) . '})', preg_quote($basename, '#')) . '$#';
             $paths[$dirname][] = [$pattern, $hash];
         }
+        $pruneFiles = [];
         foreach ($paths as $dirname => $entries) {
             foreach (new \DirectoryIterator($dirname) as $file) {
                 foreach ($entries as $entry) {
                     list($pattern, $hash) = $entry;
                     if (preg_match($pattern, $file->getFilename(), $matches) && $matches[1] !== $hash) {
-                        unlink($file->getPathname());
-                        $this->output->writeln('<comment>Deleted ' . $file->getPathname() . '</comment>');
+                        $group = sprintf(
+                            '%s/%s',
+                            basename($dirname),
+                            preg_replace('/\$.*$/', '', $file->getFilename())
+                        );
+                        if (!array_key_exists($group, $pruneFiles)) {
+                            $pruneFiles[$group] = [];
+                        }
+                        // Mark file for pruning.
+                        $pruneFiles[$group][] = new \SplFileInfo($file->getPathname());
```

Comment:
why recreating a different  object ? use `$file` directly.

Reply:
Using `$file` directly will not work outside of the foreach of `new \DirectoryIterator($dirname)`. Outside of the foreach all values of the object are set to an empty string.